### PR TITLE
Initial versal support (host)

### PR DESCRIPTION
**Pull request by larry9523**
This PR is the initial support for versal.

It added new PCIe device ID 0x5028 (will be changed to 0x5029 when we have PF swap done)
We have userpf and the following subdev driver
1) XDMA
2) mb_scheduler
3) mailbox_versal

With this initial PR, we can download XCLBIN/PDI, schedule PL and AIE kernel to run.

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c [hunk 1]
```diff
@@ -267,7 +268,7 @@ xocl_read_axlf_helper(struct xocl_drm *drm_p, struct drm_xocl_axlf *axlf_ptr)
 	}
 
 	xclbin_id = XOCL_XCLBIN_ID(xdev);
-	if (uuid_equal(xclbin_id, &bin_obj.m_header.uuid)) {
+	if (xclbin_id && uuid_equal(xclbin_id, &bin_obj.m_header.uuid)) {
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: what if on a non-versal board and xclbin_id != bin_obj uuid?

Comment (larry9523) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: Did you see any different behavior on non-versal board by this change?

Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: ok, my description is in-correct. xclbin_id is actually a pointer. I am fine with it.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c [hunk 2]
```diff
@@ -380,27 +381,35 @@ xocl_read_axlf_helper(struct xocl_drm *drm_p, struct drm_xocl_axlf *axlf_ptr)
...
+			 * since we have cleaned it up before download.
+			 */
+		}
 	}
 
-	if (!preserve_mem) {
-		int rc = xocl_init_mem(drm_p);
+	if (XOCL_DSA_IS_VERSAL(xdev)) {
+		if (drm_p->mm == NULL) {
+			rc = xocl_init_mem(drm_p, new_topology);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: can we do the same for Alveo board as well?

File: src/runtime_src/core/pcie/driver/linux/xocl/devices.h [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright (C) 2018, Xilinx Inc
+ *  Copyright (C) 2019, Xilinx Inc
```
[the remark this answers is not in the document]
Comment (larry9523) [reply src/runtime_src/core/pcie/driver/linux/xocl/devices.h]: fixed

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c (hunk 2 above)
Comment (larry9523) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: Maybe it is safe to keep the Alveo board behavior for now. Later, most likely we will remove this XOCL_DSA_IS_VERSAL here after we figuer out a way to record mem topology on versal without icap.

**Pull request merged**